src/action-docs.ts

```diff
@@ -55,40 +55,58 @@ export const defaultOptions: DefaultOptions = {
   includeNameHeader: false,
 };
 
-type ActionInputsOutputs = Record<string, ActionInput | ActionOutput>;
+type ActionInputsOutputs = Record<string, InputOutput>;
+
+enum InputType {
+  number,
+  string,
+  boolean,
+}
+
+enum InputOutputType {
+  actionInput,
+  workflowInput,
+  actionOutput,
+}
+
+const inputOutputHeaders: Record<InputOutputType, string[]> = {
+  [InputOutputType.actionInput]: ["name", "description", "required", "default"],
+  [InputOutputType.workflowInput]: [
+    "name",
+    "description",
+    "type",
+    "required",
+    "default",
+  ],
+  [InputOutputType.actionOutput]: ["name", "description"],
+};
+
+const inputOutputDefaults: Record<string, string> = {
+  description: "",
+  type: "",
+  required: "false",
+  default: '""',
+};
 
-interface ActionInput {
+interface InputOutput {
   required?: boolean;
   description?: string;
   default?: string;
-}
-
-interface ActionOutput {
-  description: string;
+  type?: InputType;
 }
 
 function createMdTable(
   data: ActionInputsOutputs,
   options: DefaultOptions,
-  type: "input" | "output",
+  type: InputOutputType,
 ): string {
   const tableData = getInputOutput(data, type);
 
   const headers = tableData.headers;
   const filler = Array(tableData.headers.length).fill("---");
 
   const result = [headers, filler]
-    .concat(
-      tableData.rows.map((line) => {
-        return line.map((elem, i) => {
-          const pretty =
-            i === 0 || i === 2 || i === 3 ? `\`${line[i]}\`` : elem;
-          const html = i === 1 ? converter.makeHtml(pretty) : pretty;
-          const htmlNoNewlines = html.replace(/(\r\n|\n|\r)/gm, " ").trim();
-          return htmlNoNewlines;
-        });
-      }),
-    )
+    .concat(tableData.rows)
```

Comment:
Seems the complex mapping is not needed anymore. Can you comment why it is removed. Or did you maybe moved it. To help reviewers you can make also on the original side of the diff comments to explain certain choices.

Reply:
The formatting is now in the `getInputOutput` function, and it's based on the column types explicitly rather than by index so that it works with action/workflow having different columns

Reply:
Yup all good, took me some time to dig in the code.